extensions/data-transfer/portability-data-transfer-generic/src/main/java/org/datatransferproject/datatransfer/generic/GenericTransferExtension.java

```diff
@@ -0,0 +1,240 @@
+import org.datatransferproject.types.common.models.calendar.CalendarContainerResource;
+import org.datatransferproject.types.common.models.media.MediaContainerResource;
+import org.datatransferproject.types.common.models.social.SocialActivityContainerResource;
+
+/**
+ * Wrapper to adapt items known to be in temp storage (e.g. BLOB data) into {@link DownloadableItem}
+ *
+ * <p>It's useful to wrap such items so upstream code can consume either known temp store'd items or
+ * items the Importer has to download itself (some MEDIA items) from the same interface.
+ */
+class CachedDownloadableItem implements DownloadableItem {
+  private String cachedId;
+  private String name;
+
+  public CachedDownloadableItem(String cachedId, String name) {
+    this.cachedId = cachedId;
+    this.name = name;
+  }
+
+  @Override
+  public String getIdempotentId() {
+    return cachedId;
+  }
+
+  @Override
+  public String getFetchableUrl() {
+    // 'url' is ID when cached
+    return cachedId;
+  }
+
+  @Override
+  public boolean isInTempStore() {
+    return true;
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+}
+
+public class GenericTransferExtension implements TransferExtension {
+
+  private Map<DataVertical, Importer<?, ?>> importerMap = new HashMap<>();
+
+  class BlobbyContainerPath {
+    private BlobbyStorageContainerResource container;
+    private String path;
+
+    public BlobbyContainerPath(BlobbyStorageContainerResource container, String path) {
+      this.container = container;
+      this.path = path;
+    }
+
+    public BlobbyStorageContainerResource getContainer() {
+      return container;
+    }
+
+    public String getPath() {
+      return path;
+    }
+  }
+
+  @Override
+  public void initialize(ExtensionContext context) {
+    JobStore jobStore = context.getService(JobStore.class);
+
+    importerMap.put(
+        BLOBS,
+        new GenericFileImporter<BlobbyStorageContainerResource>(
+            (head, om) -> {
+              List<ImportableData> results = new ArrayList<>();
+              // Search whole tree of container resource
+              Queue<BlobbyContainerPath> horizon = new ArrayDeque<>();
+              horizon.add(new BlobbyContainerPath(head, ""));
+
+              BlobbyContainerPath containerAndPath;
+              while ((containerAndPath = horizon.poll()) != null) {
+                BlobbyStorageContainerResource container = containerAndPath.getContainer();
+                String parentPath = containerAndPath.getPath();
+                String path = format("%s/%s", parentPath, container.getName());
+                // Import the current folder
+                // TODO: Consider making a POJO for mapping BlobbyFolderData and BlobbyFileData
+                ObjectNode folder = JsonNodeFactory.instance.objectNode();
+                folder.set("@type", JsonNodeFactory.instance.textNode("BlobbyFolderData"));
```

Comment:
There's some inconsistency in `@type` fields generated manually and those autogenerated for e.g. `VideoModel` - this inconsistency should be resolved